packages/admin/cms-admin/src/pages/pageTree/useExtractPages.tsx

```diff
@@ -0,0 +1,275 @@
+                                const documentWithUpdateContents = documentType.replaceTextContents?.(data.page.document, content.contents);
+
+                                if (documentWithUpdateContents && documentType.updateMutation && documentType.inputToOutput) {
+                                    await client.mutate<GQLUpdatePageMutation, GQLUpdatePageMutationVariables>({
+                                        mutation: documentType.updateMutation,
+                                        variables: {
+                                            pageId: data.page.document.id,
+                                            input: documentWithUpdateContents,
+                                            attachedPageTreeNodeId: page.id,
+                                        },
+                                        context: LocalErrorScopeApolloContext,
+                                    });
+                                }
+                            }
+
+                            const importedName = content.contents.find((item) => item.original === page.name)?.replaceWith;
+                            const importedSlug = content.contents.find((item) => item.original === page.slug)?.replaceWith.toLowerCase();
+
+                            if (importedName !== page.name || importedSlug !== page.slug) {
+                                const attachedDocument: { id?: string; type: string } = { type: page.documentType };
+
+                                if (data.page?.document) {
+                                    attachedDocument.id = data.page?.document.id;
+                                }
+
+                                await client.mutate<GQLUpdatePageNodeMutation, GQLUpdatePageNodeMutationVariables>({
+                                    mutation: updatePageNodeMutation,
+                                    variables: {
+                                        nodeId: page.id,
+                                        input: {
+                                            name: importedName && importedName !== "" ? importedName : page.name,
+                                            slug:
+                                                importedSlug && importedSlug !== "" ? transformToSlug(importedSlug ?? page.slug, locale) : page.slug,
+                                            attachedDocument,
+                                        },
+                                    },
+                                    context: LocalErrorScopeApolloContext,
+                                });
+                            }
+                        }
+                    } catch (e) {
+                        throw new Error(`Error while fetching page`);
+                    }
+                }),
+            );
+        },
+        [client, documentTypes, locale],
+    );
+
+    const getContentsFromClipboard = async (): Promise<GetContentsFromClipboardResponse> => {
```

Comment:
why build that function in a hook?

Reply:
I was using the method `getFromClipboard` of `useCopyPastePages.ts` as an inspiration